integration/tests/cook/test_cli.py

```diff
@@ -1011,6 +1012,51 @@ def test_ls_empty_root_directory(self):
         self.assertEqual(0, cp.returncode, cp.stderr)
         self.assertEqual(0, len(entries))
 
+    def test_ls_with_plugin(self):
+
+        def entry(name):
+            return cli.ls_entry_by_name(entries, name)
+
+        # Submit a job that clears the sandbox
+        cp, uuids = cli.submit("'rm -r * && rm -r .*'", self.cook_url)
+        self.assertEqual(0, cp.returncode, cp.stderr)
+        util.wait_for_job(self.cook_url, uuids[0], 'completed')
+        self.assertEqual(0, cp.returncode, cp.stderr)
+
+        # User-defined plugin to produce dummy files in the sandbox
+        with tempfile.NamedTemporaryFile(suffix='.py', delete=True) as temp:
+            path1 = str(uuid.uuid4())
+            path2 = str(uuid.uuid4())
+            plugin_code = f"""
+def dummy_ls_entries(_, __, ___):
+    return [
+        {{'path': '{path1}', 'nlink': 1, 'mode': '-rw-r--r--', 'size': 0}},
+        {{'path': '{path2}', 'nlink': 2, 'mode': 'drwxr-xr-x', 'size': 1}}
+    ]"""
+            temp.write(plugin_code.encode())
+            temp.flush()
+            config = {
+                'plugins': {
+                    "retrieve-ls-entries": {
+                        "module-name": "does_not_matter",
+                        "path": temp.name,
+                        "function-name": "dummy_ls_entries"
+                    }
+                }}
+            with cli.temp_config_file(config) as path:
+                flags = f'--config {path} --verbose'
+                cp, entries = cli.ls(uuids[0], self.cook_url, flags=flags)
+                self.assertEqual(0, cp.returncode, cp.stderr)
+                self.assertEqual(2, len(entries))
+                entry1 = entry(path1)
```

Comment:
Would it be better to just call `cli.ls_entry_by_name` instead of the one line helper function above to make it clearer that it's calling `ls_entry_by_name` and avoid shadowing `entries`?

Reply:
Definitely. Changed, thanks.